crates/ide_assists/src/handlers/convert_iter_for_each_to_for.rs

```diff
@@ -46,24 +46,25 @@ pub(crate) fn convert_iter_for_each_to_for(acc: &mut Assists, ctx: &AssistContex
     let body = closure.body()?;
 
     let stmt = method.syntax().parent().and_then(ast::ExprStmt::cast);
-    let syntax = stmt.as_ref().map_or(method.syntax(), |stmt| stmt.syntax());
```

Comment:
tbh, not a fan of or_else here -- `.syntax()` is an inline field access, it is transparent to the optimizer